cmd/meroxa/turbine/core.go

```diff
@@ -0,0 +1,97 @@
+package turbine
+
+import (
+	"context"
+	"time"
+
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/meroxa/cli/cmd/meroxa/turbine/internal"
+
+	"github.com/meroxa/turbine-core/pkg/client"
+	"github.com/meroxa/turbine-core/pkg/server"
+
+	pb "github.com/meroxa/turbine-core/lib/go/github.com/meroxa/turbine/core"
+)
+
+type Core struct {
+	grpcListenAddress string
+	builder           server.Server
+	runner            server.Server
+	client            client.Client
+}
+
+func NewCore() *Core {
+	return &Core{
+		runner:            server.NewRunServer(),
+		grpcListenAddress: internal.RandomLocalAddr(),
+		builder:           server.NewSpecBuilderServer(),
+	}
+}
+
+func (t *Core) Start(ctx context.Context) string {
+	go t.builder.RunAddr(ctx, t.grpcListenAddress)
+	return t.grpcListenAddress
+}
+
+func (t *Core) Stop() (func(), error) {
+	c, err := client.DialTimeout(t.grpcListenAddress, time.Second)
```

Comment:
This feels like a side-effect of stopping, it's actually deceiving of what it does ;-)

Reply:
May be better to shove it in Start, but have to wait for the server to actually start.

Reply:
@raulb i think there is a chance the client may try to connect before the go routine launches the server, otherwise the client may fail to connect.